python/tvm/relay/frontend/pytorch.py

```diff
@@ -1733,12 +1780,19 @@ def _convert_dtype_value(val):
                                0:"torch.unit8",
                                None:"torch.int64"} # Default is torch.int64
     if val in convert_torch_dtype_map:
-        return convert_torch_dtype_map[val]
+        return _convert_data_type(convert_torch_dtype_map[val])
     else:
         msg = "Torch data type value %d is not handled yet." % (val)
         raise NotImplementedError(msg)
 
-def _convert_data_type(input_type):
+def _convert_data_type(input_type, default_dtype=None):
+    """converts the PyTorch scalar type input_type to a TVM dtype.
+       optionally, default_dtype can be a TVM dtype that is used
+       if input_type is None (but not when it is unknown)"""
+    if input_type is None and default_dtype is not None:
+        return default_dtype
+
+    input_type = input_type.lower()
     if input_type in ["double", "torch.float64"]:
         return "float64"
     elif input_type in ["float", "torch.float32"]:
```

Comment:
Why?
float64 is not a valid input here as it will not be produced by the PyTorch JIT (which uses double).
One of the main points of this patch is to clearly separate PyTorch dtypes (which are input here) from TVM dtypes (which must not be fed into this).

Reply:
I would strongly want to avoid that. Did you find cases where we fed the wrong type here?

Reply:
I mean line 1742 (i cant select that line whle review)
input_type in ["float", "torch.float32"]: >> input_type in ["float", "float32", "torch.float32"]:
u can run the script i mentioned below. it will produce "float32" & "int64"

Reply:
You mean the the GPT2 example? It doesn't throw that for me.

Reply:
You can ignore this. this is not an issue